src/shared/sam/cli/samCliInvoker.ts

```diff
@@ -8,42 +8,168 @@
+
+            return await this.runCliCommand(samCliLocation, options, ...args)
         }
 
+        const errorResult: ChildProcessResult = {
+            exitCode: 1,
+            stdout: '',
+            stderr: '',
+            error: undefined
+        }
+        switch (validationResult.validation) {
+            case SamCliVersionValidation.VersionTooHigh:
+                const samVersionTooHighMessage = 'AWS Toolkit is out of date'
+                errorResult.error = new Error(samVersionTooHighMessage)
+                errorResult.stdout = samVersionTooHighMessage
+                errorResult.stderr = samVersionTooHighMessage
+                break
+            case SamCliVersionValidation.VersionTooLow:
+            case SamCliVersionValidation.VersionNotParseable:
+            default:
+                const samVersionTooLowMessage = 'SAM CLI is out of date'
+                errorResult.error = new Error(samVersionTooLowMessage)
+                errorResult.stdout = samVersionTooLowMessage
+                errorResult.stderr = samVersionTooLowMessage
+                break
+        }
+        // prompt will redirect to external URL for updating. We do not need to wait on this.
+        // tslint:disable-next-line: no-floating-promises
+        this._context.validator.notifyVersionIsNotValid(validationResult)
+        logger.error(errorResult.error)
+
+        return errorResult
+    }
+
+    private async runCliCommand(
+        samCliLocation: string,
+        options?: SpawnOptions,
+        ...args: string[]
+    ): Promise<ChildProcessResult>  {
         const childProcess: ChildProcess = new ChildProcess(samCliLocation, options, ...args)
 
         return await childProcess.run()
     }
+
+    private async getCliValidation(samCliLocation: string): Promise<SamCliVersionValidatorResult> {
+        const cliStat = await this._context.cliUtils.stat(samCliLocation)
+        if (!this._context.cliInfo.lastModified || cliStat.mtime !== this._context.cliInfo.lastModified) {
+            this._context.cliInfo.lastModified = cliStat.mtime
+            this._context.cliInfo.info = await this.getInfo(samCliLocation)
+        }
+        if (!this._context.cliInfo.info) {
+            this._context.cliInfo.info = await this.getInfo(samCliLocation)
+        }
+
+        return await this._context.validator.getCliValidationStatus(this._context.cliInfo.info.version)
+    }
+
+    private async getInfo(samCliLocation: string): Promise<SamCliInfoResponse> {
+        const logger: Logger = getLogger()
+
+        const { error, exitCode, stderr, stdout }: ChildProcessResult =
+            await this.runCliCommand(samCliLocation, undefined, '--info')
+
+        if (exitCode === 0) {
+            const response = this.convertOutput(stdout)
+            if (!!response) {
+                return response
+            }
+
+            throw new Error('SAM CLI did not return expected data')
+        }
+
+        console.error('SAM CLI error')
```

Comment:
I realize this code was moved over, consider using the logger now that we have it.
If you choose to do this, please also log the command that was run.

Reply:
We log underneath. I'll go ahead and remove this (we also have this same setup in every other SAM CLI command, should I do the same to those as well?)

Reply:
Where are they logged underneath?
The other commands can be adjusted in a separate PR.

Reply:
Err, logged underneath these lines of code. These messages (both the console.error statements and the error itself) were present before the initial logger changes, I just co-opted the error itself for the logger statement.
```
        const err =
            new Error(`sam --info encountered an error: ${error && error.message ? error.message : stderr || stdout}`)
        logger.error(err)
        throw err
```

Reply:
Ah okay. Would it be more useful to unconditionally log all of the fields? That statement has error.message potentially mask out stderr and stdout.

Reply:
How's this?
```
        const err = new Error(
`sam --info encountered an error: ${error}
    ${error && error.message ? 'message: ' + error.message : ''}
    stderr : ${stderr}
    stdout: ${stdout}`
        )
```